### ISSUE TITLE
count(*) with ORDER BY returns empty result if no rows exist

### ISSUE THREAD

**Issue by mmacfadden** [orientechnologies/orientdb#9110]
## Issue:

When performing a count with an ORDER BY. If no rows match, instead of getting a result with a count field set to 0, we get an empty result set.  This is similar to #8280 and #8621 which I had previously reported.

## Expected behavior  
A count result of 0 should be returned.

## Actual behavior  
An empty result set is returned.

## Steps to reproduce

Create the schema:
```SQL
CREATE CLASS Test;
CREATE Property Test.id String;
``` 

### Test 1 (Works)
```SQL
SELECT count(*) as count FROM Test;
```

```JSON
    "result": [
        {
            "@type": "d",
            "@version": 0,
            "count": 0,
            "@fieldTypes": "count=l"
        }
    ]
```

### Test 2 (Doesn't Work)
```SQL
SELECT count(*) as count FROM Test ORDER BY id ASC;
```

```JSON
    "result": [
        
    ]
```

**Comment by luigidellaquila** [comment]
Hi @mmacfadden 

this query does not make much sense semantically, but I think it's worth fixing anyway.
I'm pushing a fix now, it will be released with v 3.0.28

Thanks

**Comment by mmacfadden** [comment]
@luigidellaquila Thanks! You are correct, it is a little non-sensical.  This came up because we were working with pagination and we have a utility that essentially takes your original query (that might have SKIP and LIMIT in it) and builds a second query that does the count to get the total result.  The order by was coming from the original query!
